tests/generative/variable_strategies.py

```diff
@@ -58,6 +61,53 @@ def variable(patient_tables, event_tables, schema, value_strategies):
     #     permitted to return a one-row-per-patient series, because such series can always be
     #     composed a many-rows-per-patient series; so there are series strategy functions that,
     #     always or sometimes, ignore the frame argument.
+
+    # Decorators
+    #
+    # These two decorators, one for series strategies and one for table strategies, force
+    # hypothesis to return a terminating node if the current test depth is too deep.
+    # Otherwise, the generated graph can continue forever, and will eventually hit the
+    # hypothesis limit (100) and will be abandoned. This results in too many invalid examples,
+    # which triggers the too-many-filters healthcheck.
+    #
+    # If the max limit is set high - e.g. if we always let it go to 100 and then return our
+    # default terminating node, generating the examples takes a really long time.
+    #
+    # Setting the depth at which a table strategy bails too low means that hypothesis takes
+    # too long to shrink examples.
+    #
+    # The defaults are therefore set, somewhat arbitrarily, to 30 for series and 75 for tables.
```

Comment:
I bet that was fun figuring out.

Reply:
I'm not convinced about those numbers, but they can be modifed with env variables so we can try out different values if it turns out to take forever with the overnight ones or something